core/services/workflows/syncer/workflow_registry.go

```diff
@@ -221,22 +246,28 @@ func (w *workflowRegistry) Start(_ context.Context) error {
 				return
 			}
 
-			w.lggr.Debugw("Loading initial workflows for DON", "DON", don.ID)
-			loadWorkflowsHead, err := w.loadWorkflows(ctx, don, reader)
-			if err != nil {
-				// TODO - this is a temporary fix to handle the case where the chainreader errors because the contract
-				// contains no workflows.  To track: https://smartcontract-it.atlassian.net/browse/CAPPL-393
-				if !strings.Contains(err.Error(), "attempting to unmarshal an empty string while arguments are expected") {
-					w.lggr.Errorw("failed to load workflows", "err", err)
-					return
-				}
+			// Start goroutines to gather changes from Workflow Registry contract
+			switch w.syncStrategy {
+			case SyncStrategyEvent:
+				w.syncUsingEventStrategy(ctx, don, reader)
```

Comment:
This is going to block I think -- you need to do dispatch a top-level goroutine

i.e. go w.syncUsingEventStrategy

It'll block because you'll do the synchronous fetching of the initial metadata put it on the channel, but that channel will only get read from on line 264